File: src/main/java/cz/cvut/kbss/changetracking/annotation/Audited.java
package cz.cvut.kbss.changetracking.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * {@code @Audited} is used to enable change tracking for an entity class.
 * <p>
 * {@link cz.cvut.kbss.changetracking.strategy.entity.EntityStrategy#checkClassSupported(Class)} implementations are
 * encouraged to check if the examined class has this annotation.
 */
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.TYPE})
public @interface Audited {
}

File: src/main/java/cz/cvut/kbss/changetracking/annotation/Audited.kt
package cz.cvut.kbss.changetracking.annotation

/**
 * `@Audited` is used to enable change tracking for an entity class.
 *
 * [cz.cvut.kbss.changetracking.strategy.entity.BaseEntityStrategy.checkClassSupported] implementations are
 * encouraged to check if the examined class has this annotation.
 */
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS)
annotation class Audited
